src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -840,10 +841,35 @@ def save_as_table(
         child: SnowflakePlan,
         source_plan: Optional[LogicalPlan],
         use_scoped_temp_objects: bool,
-        is_generated: bool,  # true if the table is generated internally
-        child_attributes: List[Attribute],
+        creation_source: TableCreationSource,
+        child_attributes: Optional[List[Attribute]],
     ) -> SnowflakePlan:
+        """Returns a SnowflakePlan to materialize the child plan into a table.
+
+        Args:
+            table_name: fully qualified table name
+            column_names: names of columns for the table
+            mode: APPEND, TRUNCATE, OVERWRITE, IGNORE, ERROR_IF_EXISTS
+            table_type: temporary, transient, or permanent
+            clustering_keys: list of clustering columns
+            comment: comment associated with the table
+            child: the SnowflakePlan that is being materialized into a table
+            source_plan: the source plan of the child
+            use_scoped_temp_objects: should we use scoped temp objects
+            creation_source: the source of the api that is creating the table
+            child_attributes: child attributes will be none in the case of large query breakdown
+                where we use ctas query to create the table which does not need to know the column
+                metadata.
+        """
+        is_generated = creation_source in (
+            TableCreationSource.CACHE_RESULT,
+            TableCreationSource.LARGE_QUERY_BREAKDOWN,
+        )
         if is_generated and mode != SaveMode.ERROR_IF_EXISTS:
+            # an internally generated save_as_table comes from two sources:
+            #   1. df.cache_result: plan is created with create table + insert statement
+            #   2. large query breakdown: plan is created using a CTAS statement
+            # For these cases, we must use mode ERROR_IF_EXISTS
```

Comment:
let's also add a check that if child_attribute is None, it can only be for the case when creation source is TableCreationSource.LARGE_QUERY_BREAKDOWN